.cirrus.yml

```diff
@@ -0,0 +1,121 @@
+common_steps_template: &COMMON_STEPS_TEMPLATE
+  install_prerequisites_script: |
+    set -uexo pipefail
+    if [ "$OS_NAME" == "linux" ]; then
+      packages="git-core make g++ gdb curl libcurl3 tzdata zip unzip xz-utils"
+      if [ "$MODEL" == "32" ]; then
+        dpkg --add-architecture i386
+        packages="$packages g++-multilib libcurl3-gnutls:i386"
+      fi
+      if [ "${DMD:0:4}" == "gdmd" ]; then
+        packages="$packages sudo software-properties-common wget"
+      fi
+      apt-get -q update
+      apt-get install -yq $packages
+    elif [ "$OS_NAME" == "darwin" ]; then
+      # required for install.sh
+      brew install gnupg
+    elif [ "$OS_NAME" == "freebsd" ]; then
+      pkg install -y git gmake bash
+      rm /usr/bin/make
+      ln -s /usr/local/bin/gmake /usr/bin/make
+    fi
+    # create a `dmd` symlink to the repo dir, necessary for druntime/Phobos
+    ln -s $CIRRUS_WORKING_DIR ../dmd
+  install_host_compiler_script: |
+    source ci.sh
+    # kludge
+    if [ "${DMD:0:4}" == "gdmd" ]; then export DMD="gdmd"; fi
```

Comment:
Why not update the `ci.sh` script?

```diff
diff --git a/ci.sh b/ci.sh
index 14c4f3ee2..e8909e6d4 100755
--- a/ci.sh
+++ b/ci.sh
@@ -163,7 +163,7 @@ download_install_sh() {
 }
 
 install_d() {
-  if [ "${DMD:-dmd}" == "gdc" ] || [ "${DMD:-dmd}" == "gdmd" ] ; then
+  if [[ "${DMD:-dmd}" =~ "gdc" ]] || [[ "${DMD:-dmd}" =~ "gdmd" ]] ; then
     export DMD=gdmd-${GDC_VERSION}
     if [ ! -e ~/dlang/gdc-${GDC_VERSION}/activate ] ; then
         sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
```

Reply:
I tried not to touch it for now, but it could definitely need an overhaul to be less Semaphore-specific. E.g. here, I had to install wget because it doesn't curl (and sudo as well). Also, it looks like the gdmd wrapper could use a revision as well; I firstly tried to symlink `gdmd` to `gdmd-9`, but that doesn't work, because the script parses its own name to get to the version; I guess there's a way to resolve the symlink to overcome stupid things like that.